lib/libc/math/lib_asin.c

```diff
@@ -64,6 +64,14 @@ double asin(double x)
 
 	y = 0;
 
+	if (isnan(x)) {
+		return NAN;
+	}
+
+	if (x < -1.0 || x > 1.0) {
```

Comment:
Try to reduce the redundant if loops by clubbing the conditions.

Reply:
Modified